pillow_jxl/JpegXLImagePlugin.py

```diff
@@ -33,9 +33,13 @@ def _open(self):
         if self.jpeg:
             with Image.open(BytesIO(self._data)) as im:
                 self._data = im.tobytes()
-        self._size = (self._jxlinfo.width, self._jxlinfo.height)
-        self.rawmode = self._jxlinfo.mode
-        self.info["icc_profile"] = icc_profile
+                self._size = im.size
+                self.rawmode = im.mode
+                self.info = im.info
+        else:
+            self._size = (self._jxlinfo.width, self._jxlinfo.height)
+            self.rawmode = self._jxlinfo.mode
+            self.info["icc_profile"] = icc_profile
```

Comment:
According to [CI log](https://github.com/Isotr0py/pillow-jpegxl-plugin/actions/runs/11204228897/job/31150927760?pr=70#step:8:53), Seems that `icc_profile` is missing here for reconstructed jpeg.